MonoGame.Framework/Color.cs

```diff
@@ -273,14 +288,43 @@ public Color(int r, int g, int b)
         public Color(int r, int g, int b, int alpha)
         {
             _packedValue = 0;
-            R = (byte)MathHelper.Clamp(r, Byte.MinValue, Byte.MaxValue);
-            G = (byte)MathHelper.Clamp(g, Byte.MinValue, Byte.MaxValue);
-            B = (byte)MathHelper.Clamp(b, Byte.MinValue, Byte.MaxValue);
-            A = (byte)MathHelper.Clamp(alpha, Byte.MinValue, Byte.MaxValue);
+            if (((r | g | b | alpha) & 0xFFFFFF00) != 0)
+            {
+                R = (byte)MathHelper.Clamp(r, Byte.MinValue, Byte.MaxValue);
+                G = (byte)MathHelper.Clamp(g, Byte.MinValue, Byte.MaxValue);
+                B = (byte)MathHelper.Clamp(b, Byte.MinValue, Byte.MaxValue);
+                A = (byte)MathHelper.Clamp(alpha, Byte.MinValue, Byte.MaxValue);
+            }
+            else
+            {
+                R = (byte)r;
+                G = (byte)g;
+                B = (byte)b;
+                A = (byte)alpha;
+            }
+        }
+
+        /// <summary>
+        /// Constructs an RGBA color from scalars representing red, green, blue and alpha values.
+        /// </summary>
+        /// <remarks>
+        /// This overload sets the values directly without clamping, and may therefore be faster than the other overloads.
+        /// </remarks>
+        /// <param name="r"></param>
+        /// <param name="g"></param>
+        /// <param name="b"></param>
+        /// <param name="alpha"></param>
+        public Color(byte r, byte g, byte b, byte alpha)

```

Comment:
You are suggesting that `new Color(128,128,128,128)` will always call the `int` version of the constructor.  This would mean you would have to either put the values into a temporary or explicitly cast like `new Color((byte)128,(byte)128,(byte)128,(byte)128)`.

If this is correct... and I think it is... should we instead consider ` Color.FromBytes(byte,byte,byte,byte)` for this optimized construction method?

```

```